addons/account/tests/test_early_payment_discount.py

```diff
@@ -498,3 +498,69 @@ def assert_tax_totals(document, expected_values):
                     }
                 ],
             })
+
+    def test_mixed_epd_with_tax_included(self):
+        def assert_tax_totals(document, expected_values):
+            main_keys_to_ignore = {
+                'formatted_amount_total', 'formatted_amount_untaxed', 'display_tax_base', 'subtotals_order'}
+            group_keys_to_ignore = {'group_key', 'tax_group_id', 'tax_group_name',
+                                    'formatted_tax_group_amount', 'formatted_tax_group_base_amount'}
+            subtotals_keys_to_ignore = {'formatted_amount'}
+            to_compare = document
+            for key in main_keys_to_ignore:
+                del to_compare[key]
+            for key in group_keys_to_ignore:
+                for groups in to_compare['groups_by_subtotal'].values():
+                    for group in groups:
+                        del group[key]
+            for key in subtotals_keys_to_ignore:
+                for subtotal in to_compare['subtotals']:
+                    del subtotal[key]
+            self.assertEqual(to_compare, expected_values)
+
+        self.env.company.early_pay_discount_computation = 'mixed'
```

Comment:
```suggestion
        self.company_data['company'].early_pay_discount_computation = 'mixed'
```
If we know we are going to use this company, it's better to select this one directly so it doesn't depend on `env`